src/utils/COMETAdapter.cpp

```diff
@@ -0,0 +1,577 @@
+    vector<ResidueModification> fixed_modifications = getModifications_(fixed_modifications_names);
+    for (vector<ResidueModification>::const_iterator it = fixed_modifications.begin(); it != fixed_modifications.end(); ++it)
+    {
+      String AA = it->getOrigin();
+      if ((AA!="N-term") && (AA!="C-term"))
+      {
+      const Residue* r = ResidueDB::getInstance()->getResidue(AA);
+      String name = r->getName();
+      os << "add_" << r->getOneLetterCode() << "_" << name.toLower() << " = " << it->getDiffMonoMass() << endl;
+      }
+      else
+      {
+      os << "add_" << AA.erase(1,1) << "_peptide = " << it->getDiffMonoMass() << endl;
+      }
+    }
+
+    // COMET_ENZYME_INFO _must_ be at the end of this parameters file
+    os << "[COMET_ENZYME_INFO]" << "\n";
+    os << "0.  No_enzyme              0      -           -" << "\n";
+    os << "1.  Trypsin                1      KR          P" << "\n";
+    os << "2.  Trypsin/P              1      KR          -" << "\n";
+    os << "3.  Lys_C                  1      K           P" << "\n";
+    os << "4.  Lys_N                  0      K           -" << "\n";
+    os << "5.  Arg_C                  1      R           P" << "\n";
+    os << "6.  Asp_N                  0      D           -" << "\n";
+    os << "7.  CNBr                   1      M           -" << "\n";
+    os << "8.  Glu_C                  1      DE          P" << "\n";
+    os << "9.  PepsinA                1      FL          P" << "\n";
+    os << "10. Chymotrypsin           1      FWYL        P" << "\n";
+  }
+
+  ExitCodes main_(int, const char**)
+  {
+    //-------------------------------------------------------------
+    // parsing parameters
+    //-------------------------------------------------------------
+
+    String inputfile_name = getStringOption_("in");
+    writeDebug_(String("Input file: ") + inputfile_name, 1);
+    if (inputfile_name == "")
+    {
+      writeLog_("No input file specified. Aborting!");
+      printUsage_();
+      return ILLEGAL_PARAMETERS;
+    }
+
+    String out = getStringOption_("out");
+    writeDebug_(String("Output file___real one: ") + out, 1);
+    if (out == "")
+    {
+      writeLog_("No output file specified. Aborting!");
+      printUsage_();
+      return ILLEGAL_PARAMETERS;
+    }
+
+
+    //-------------------------------------------------------------
+    // reading input
+    //-------------------------------------------------------------
+
+    String db_name(getStringOption_("database"));
+    if (!File::readable(db_name))
+    {
+      String full_db_name;
+      try
+      {
+        full_db_name = File::findDatabase(db_name);
+      }
+      catch (...)
+      {
+        printUsage_();
+        return ILLEGAL_PARAMETERS;
+      }
+      db_name = full_db_name;
+    }
+
+    //tmp_dir
+    //const String tmp_dir = QDir::toNativeSeparators((File::getTempDirectory() + "/").toQString());
+    const String tmp_dir = OpenMS::File::getTempDirectory() + "/";
```

Comment:
I've recently added the functions `makeTempDirectory_()` and `removeTempDirectory_` to TOPPBase. Just use those!
(They are not in the File class because `removeTempDirectory_` uses the debug level from TOPPBase.)